### PR TITLE
[JJBB] Load testing MBP

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Enable MBP for the load testing pipeline. It listen for PRs and master branch but builds won't be triggered automatically but manually using the UI to customise the parameters.

### CODE REVIEW

File: .ci/jobs/apm-agent-java-load-mbp.yml
Comment (v1v) [inline .ci/jobs/apm-agent-java-load-mbp.yml:18]: Filter only master and PRs

File: .ci/jobs/apm-agent-java-load-mbp.yml
Comment (v1v) [inline .ci/jobs/apm-agent-java-load-mbp.yml:16]: Avoid GH status checks

File: .ci/jobs/apm-agent-java-load-mbp.yml
Comment (v1v) [inline .ci/jobs/apm-agent-java-load-mbp.yml:26]: Avoid running when a PR is created or a merge to master happens automatically

File: .ci/load/Jenkinsfile
Comment (v1v) [inline .ci/load/Jenkinsfile:43]: Ensure the first time contributor check happens

File: .ci/jobs/apm-agent-java-load-mbp.yml
Comment (SylvainJuge) [reply .ci/jobs/apm-agent-java-load-mbp.yml:18]: Since the repository is the agent repository, not all PRs are related to load-testing.
- could we have a way to filter those, maybe on branch/PR name pattern ?

For example, if we want to filter on branch name, would having `(master|wip-load-testing-.*)` instead allow to have:
- one instance of the job for `master`
- one instance of the job for each branch matching pattern `wip-load-testing-*`.

One very minor downside of relying on branch name and not on PR would be to require pushing to the upstream repository.

Comment (v1v) [reply .ci/jobs/apm-agent-java-load-mbp.yml:18]: Let me explain a bit better:

The requirement we had in mind was to support the run of PRs that are related to the load testing, for such we took in consideration:
- there will no be automated builds for any PRs, or commits into the master branch.
- The only way to interact with it would be manually -> [apm-agent-java-load-mbp](https://apm-ci.elastic.co/job/apm-agent-java/job/apm-agent-java-load-mbp/) (once the job is created)

**Review by cachedout — APPROVED** [review]

**Comment by cachedout** [comment]
@SylvainJuge Go ahead and merge this once you've had a chance to review it. Once merged, a webhook should pick it up and create a new pipeline after a few minutes which should work for you to test your changes with. Let me or @v1v if you have any issues. Thanks!

**Review by SylvainJuge — APPROVED** [review]

**Pull request merged**
